app/src/main/java/com/yterletskyi/happyfriend/features/pin/data/PinCodeController.kt

```diff
@@ -0,0 +1,49 @@
+package com.yterletskyi.happyfriend.features.pin.data
+
+import android.content.SharedPreferences
+import com.yterletskyi.happyfriend.common.LifecycleComponent
+
+interface PinCodeController : LifecycleComponent {
+    val pinCode: PinCode
+}
+
+interface InternalPinCodeController : PinCodeController {
+    fun getPin(): PinCode?
+    fun savePinCode(pinCode: PinCode?)
+}
```

Comment:
let's merge the `PinCodeController`  and `InternalPinCodeController` into one interface (PinCodeController). I don't think we need to have them separate, or do we?

Also, I think  just 
```
fun getPin(): PinCode?
fun savePinCode(pinCode: PinCode?)
```

will be great and enough for us. 

Why do we need both `val pinCode: PinCode` and `fun getPin(): PinCode?` -- what is the difference between them?